### PR TITLE
chore: add script to test e2e bitcode transform + compilation + execute on qpu

### PR DESCRIPTION
**Pull request by nilslice**
This PR adds a simple script / cargo-make command to run a configurable e2e test of IR transformation + compilation to executable format + program execution.

_(Added to the README)_

> For a full end-to-end integration test against both `qvm` and QPUs, you can use:
> 
> ```sh
> cargo make release-test-e2e <TARGET> <PROGRAM>
> # TARGET = one of: qvm, Aspen-11, Aspen-M-1
> # PROGRAM = path to an LLVM bitcode (.bc) file to transform and compile into executable format
> ```
> 
> ```sh
> cargo make release-test-e2e
> # defaults to:
> # cargo make release-test-e2e qvm tests/fixtures/programs/reduction.bc
> # use any of {qvm,Aspen-11,Aspen-M-1}, assuming you can access the QPUs.
> 
> # using the default test program
> cargo make release-test-e2e Aspen-11
> 
> # using a different test program (note: target must be provided, args are positional)
> cargo make release-test-e2e Aspen-11 ./input.bc 
> ```

### CODE REVIEW

File: scripts/release/test-e2e.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    ;;
+esac
+
+echo "Transforming: '${TEST_INPUT}' -> '${DIST_DIR}/output.bc'"
+echo "Compiling: '${DIST_DIR}/program'"
+
+# transform and compile the program, then run against specified target
+case $OS in
+    "darwin"|"linux")
+        if [ $OS = "darwin" ]; then
```
Comment (notmgsk) [inline scripts/release/test-e2e.sh:53]: Does this modify the global environment? i.e. do the changes to `LD_LIBRARY_PATH` persist after the script is ran? If so, might want to make those changes temporary.

Comment (nilslice) [reply scripts/release/test-e2e.sh:54]: It modifies the current shells environment, and restoring that is not a bad idea. Added!

File: README.md [hunk 2]
```diff
@@ -48,12 +48,35 @@ NO_C_SDK_BUILD=1 NO_QIR_SDK_BUILD=1 NO_HELPER_LIB_BUILD=1 cargo make release-qui
...
+# use any of {qvm,Aspen-11,Aspen-M-1}, assuming you can access the QPUs.
+
+# using the default test program
+cargo make release-test-e2e Aspen-11
+
+# using a different test program (note: target must be provided, args are positional)
+cargo make release-test-e2e Aspen-11 ./input.bc 
 ```
 
 ## Examples
```
Comment (kalzoo) [inline README.md:82]: I wonder if the example, or (if it's too long) a shorter example, should be the leading bit in the readme as the sell for people who may not be interested in its development or testing just yet.

**Review by kalzoo — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (nilslice) [reply README.md:82]: Good call! I think this example should return some more interesting data if we lead with it that way. I'll add something in another PR to better demonstrate & update the readme.

**Pull request merged**